### ISSUE TITLE
`no-useless-range` fail to escape some characters

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#104]
Here are some character classes that will be fixed incorrectly:

```js
/[,--b]/ -> /[,-b]/
/[\c-d]/ -> /[\cd]/
```

These examples are very artificial but should be fixed at some point. This is a low-priority bug.
